BLang/Error/Syntax.cs

```diff
@@ -0,0 +1,316 @@
+﻿using BLang.Utils;
+using System.Diagnostics;
+using System.Reflection.Metadata.Ecma335;
+
+namespace BLang.Error
+{
+    public class InvalidTokenAtFileLevel : ParseError
+    {
+        public InvalidTokenAtFileLevel(ParserContext context) : base(context)
+        {
+        }
+
+        public override eErrorLevel Level => eErrorLevel.CriticalError;
+
+        public override eParseError ErrorType => eParseError.UnexpectedTokenAtFileLevel;
+
+        protected override string Message => "A file must include only a list of imports followed by a module block";
+
+        protected override bool ChildRecoverFromError()
+        {
+            // Cannot recover from this error.
+            // Maybe we can eventually insert the right token into the stream to put it into context
+            // but maybe that's not worth it.
+
+            // Or a simpler way to do it would be to pull tokens off until we find one that matches something that's 
+            // expected at the file level.
+            return false;
+        }
+    }
+
+    public class UnexpectedToken : ParseError
+    {
+        public UnexpectedToken(ParserContext context)
+            : base(context)
+        {
+        }
+
+        public override eErrorLevel Level => eErrorLevel.Error;
+
+        public override eParseError ErrorType => eParseError.UnexpectedToken;
+
+        protected override string Message
```

Comment:
it might be better to use a function here instead of a variable as they perform the same, but the variables arent necessarily expected to do large sequences of code in getters/setters